helix-core/src/main/java/org/apache/helix/spectator/RoutingTableProvider.java

```diff
@@ -290,7 +430,23 @@ public Object removeRoutingTableChangeListener(
    */
   public List<InstanceConfig> getInstancesForResource(String resourceName, String partitionName,
       String state) {
-    return _routingTableRef.get().getInstancesForResource(resourceName, partitionName, state);
+    String key = getRoutingTableKey();
+    return _routingTableRef.get(key).get().getInstancesForResource(resourceName, partitionName,
+        state);
+  }
+
+  public List<InstanceConfig> getInstancesForResource(String resourceName, String partitionName,
```

Comment:
Same here. If not in the design, let's hold this for now.

Reply:
Removed.